### PR TITLE
Restaurant UI Base implementations

### PR DESCRIPTION
**Pull request by kxyoke**
Important/global changes:
- res promotions are stored in the form `DEFAULT:min_amount;absolute|percent;discount` (see `.../client/src/components/restaurant_staff/components/utils/utils.js#getDefaultPromoDescProps`
- food schema imagepath changed to TEXT
- Orders schema added `is_prepared`

Base implementations:
- [x] profile
- [x] edit profile
- [x] menu
- [x] edit menu
- [x] reviews
- [x] orders
- [x] promos: add, edit (,delete (don't allow so still in db?)
- [x] promo history [tent]

Next PR(s):
- orders: change order status (implement both axios request and ui update)
- beautified implementations
   - home page as dashboard (current orders in scrollpane + add promo add menu shortcuts etc)
- improved capabilities + enhance usability

Update to requirements:
- [x] Input form validation: profile

### CODE REVIEW

**Comment by kxyoke** [comment]
New issues:
- [ ] Menu should support view by category [all, __, __, __] (so need to query restaurant's categories.)
- [ ] Retrieval order by alphabetical order, timestamp, etc [food items, orders, reviews, promo]
- [ ] Promo table add check [is fds || (is res && format is in res format -- LIKE and regex)

**Comment by kxyoke** [comment]
Remember to add necessary query transformation in server before responding.

**Comment by kxyoke** [comment]
Issue:
- [ ] Fix image default

**Comment by kxyoke** [comment]
- [ ] Add loaders? in case; but not v impt

File: client/src/components/restaurant_staff/components/res_menu.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        }
+    }
+
+    return (
+      <div>
+        {isLoading?
+        <p>Halo plz wait</p>
+        : 
+        <div>
+          <Tab panes={makePanes()} onTabChange={ (e,tab) => changeTabToIdx(tab.activeIndex) } />
```
Comment (kxyoke) [inline client/src/components/restaurant_staff/components/res_menu.js:161]: @ShuTingY can consider using semantic-ui-react tabs for browsing food items by categories.

File: client/src/components/restaurant_staff/components/res_menuItemForm.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        if (newCategory == cat_others) {
+            setShowOtherCategory(true)
+        } else {
+            setShowOtherCategory(false)
+        }
+    }, [newCategory])
+
+    function validate(withCat) {
+        const currencyRegex = /^[0-9]\d*(?:(\.\d{0,2})?)$/;
+        const positiveIntRegex = /^[1-9]\d*$/;
```
Comment (kxyoke) [inline client/src/components/restaurant_staff/components/res_menuItemForm.js:55]: for some reason, `Number.isInteger` doesn't work for me;
y'all can use this regex for input validation also, or alternatively respond from server with the error and handle it

File: server/sql/pgsqlSetup/schema.sql [hunk 3]
```diff
@@ -27,23 +28,29 @@ CREATE TABLE Restaurants (
...
+    FOREIGN KEY (category) REFERENCES FoodCategories 
+    -- NO ACTION ON CASCADE?
 );
 
-
 CREATE TABLE MenuItems (
-    res_id      TEXT,
-    food_id     TEXT,
+    res_id      TEXT NOT NULL,
+    food_id     TEXT PRIMARY KEY,
```
[the remark this answers is not in the document]
Comment (kxyoke) [reply server/sql/pgsqlSetup/schema.sql]: ahh cus i was thinking our `food_id` is unique across the entire fds system (also in a sense if the fds system/managers want to manage the foods directly they can do so without the notions of restaurants too (eg. description should be censored etc.))

but this is different from what our ER showed also ah
our `fooditem` is not dependent on `menuitem` but instead the other way round also though

so should we 
1) also change our ER diagram, 
2) leave it and have our schema reflect differently but specify that menuitem must be identified by res_id and food_id (such that semantically food_id is not unique across restaurants),
3) or change our schema to let menuitem own the entire big table so it follows the ER entirely?

File: server/sql/pgsqlSetup/setup.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+\i ./schema.sql
+
+INSERT INTO FoodCategories(category) VALUES ('Others');
```
[the remark this answers is not in the document]
Comment (kxyoke) [reply server/sql/pgsqlSetup/setup.sql:3]: i placed it here cus it is a compulsory piece of data on the db, but `init.sql` atm contains unverified data that we might change
we leave it here for now? then when we actually modify the init.sql we add this line to init.sql

[the remark this answers is not in the document]
Comment (ShuTingY) [reply server/sql/pgsqlSetup/setup.sql:3]: Okay

File: server/sql/pgsqlSetup/schema.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShuTingY) [reply server/sql/pgsqlSetup/schema.sql]: I think put res_id and food_id as primary key for menueItem is better as it better representation of the menu? 
I though current schema is letting the menu item own the entire big table? Or you mean combine the food table with the menu item so we just have a menuitems table? I think based on what we have now, they can be combine as food_id will only  be used by 1 restaurant? 

Anyway, you can change the schema if you want to better fit the requirements . Cause previously,  I was thinking that different restaurants’s menu can use the same food_id if they are selling the same food. Like chicken rice, fish soup.....

**Pull request merged**